src/protocols/bdx/BdxMessages.h

```diff
@@ -0,0 +1,346 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file defines structures and utility methods for working with BDX
+ *      messages, mainly for writing to and reading from PacketBuffers.
+ */
+
+#pragma once
+
+#include <support/BitFlags.h>
+#include <support/BufBound.h>
+#include <support/CodeUtils.h>
+#include <system/SystemPacketBuffer.h>
+
+namespace chip {
+namespace BDX {
+
+enum TransferControlFlags : uint8_t
+{
+    // first 4 bits reserved for version
+    kSenderDrive   = (1U << 4),
+    kReceiverDrive = (1U << 5),
+    kAsync         = (1U << 6),
+};
+
+enum RangeControlFlags : uint8_t
+{
+    kDefLen      = (1U),
+    kStartOffset = (1U << 1),
+    kWiderange   = (1U << 4),
+};
+
+/*
+ * A structure for representing a SendInit or ReceiveInit message (both contain
+ * identical parameters).
+ */
+struct TransferInit
+{
+    /**
+     * @brief
+     *  Write the message fields to a buffer using the provided BufBound.
```

Comment:
nit: "Write the message fields....." . Maybe the method name should be 'Write' then.

Below the description of "Parse data from an ...." does result in a 'Parse' method name

Reply:
Ok, I was considering changing this to "WriteToBuf" so I can go ahead and do that